gems/smithy-client/lib/smithy-client/managed_file.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Client
+    # This utility class is used to track files opened by Smithy::Client.
+    # @api private
+    class ManagedFile < File
```

Comment:
I'm not sure I really understand the purpose of this class.  In the response target handler we could just check for !closed? instead of open?

Reply:
The method really doesn't matter. We wrap string targets (path names) with a managed file so that we know we can unlink files if it's a managed file type. If we are otherwise given a file, we don't want to unlink.